### PR TITLE
Magn 6084

### PR DESCRIPTION
**Pull request by ptierney**
This pull request fixes MAGN-6084

It adds a new node, ExportToSAT, that has a dropdown menu specifying the unit type in the SAT file.

It adds a XAML for the node layout itself, as well as a NodeModel to specify the logic

@Randy-Ma PTAL

### CODE REVIEW

File: src/Libraries/GeometryUI/GeometryUI.csproj [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    <Optimize>false</Optimize>
+    <OutputPath>..\..\..\bin\AnyCPU\Debug\nodes\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (Randy-Ma) [inline src/Libraries/GeometryUI/GeometryUI.csproj]: The output need to be adjusted

File: src/Libraries/GeometryUIWpf/GeometryUIWpf.csproj [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    <Optimize>false</Optimize>
+    <OutputPath>..\..\..\bin\AnyCPU\Debug\nodes\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (Randy-Ma) [inline src/Libraries/GeometryUIWpf/GeometryUIWpf.csproj]: The output need to be adjusted

File: src/Libraries/GeometryUI/ExportWithUnits.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            SelectedExportedUnit = ConversionUnit.Feet;
+            SelectedExportedUnitsSource =
+                Conversions.ConversionMetricLookup[ConversionMetricUnit.Length];
+
+            AssociativeNode geometryNode = new ArrayNode();
+            AssociativeNode stringNode = new StringNode();
+            InPortData.Add(new PortData("geometry", "Geometry to export into a SAT file.", geometryNode));
+            InPortData.Add(new PortData("filePath", "File to export the geometry to.", stringNode));
+            OutPortData.Add(new PortData("string", "The file path of the exported file. Note this may change from the input in it contains non-ASCII characters."));
```
Comment (Randy-Ma) [inline src/Libraries/GeometryUI/ExportWithUnits.cs]: The strings used in this function need localization support.

File: test/Libraries/GeometryUI/GeometryUI.csproj [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    <RootNamespace>GeometryUI</RootNamespace>
+    <AssemblyName>GeometryUI</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
```
Comment (Randy-Ma) [inline test/Libraries/GeometryUI/GeometryUI.csproj]: the output folder

**Comment by Randy-Ma** [comment]
Sorry, I see you have updated this already.

**Comment by Randy-Ma** [comment]
@ptierney 
It generally looks good to me. Few more suggestions:
1). If there are some test cases, it will be great!
2). Can we decide the units based on the environment? Users may not know what the exact units for the geometries. Let's say if the geometries are in feet, what will it mean if users choose meters? 
3). As discussed, the other version of the geometry binaries need to be updated as well. (maybe in another pull request).

**Comment by Randy-Ma** [comment]
@ptierney 
One more comment:
For the newly added projects, AssemblySharedInfo.cs need to be added and duplicated attributes need to be removed from the original AssemblyInfo.cs.

**Comment by Randy-Ma** [comment]
Sorry for being nagging. :)

The test project name is better to be "GeometryUITests" other than "GeometryUI". But I think it is better to add the test category in the WorkflowTests project.

**Comment by ptierney** [comment]
No problem, this makes sense, I've move the test over to WorkflowTest project

**Comment by Randy-Ma** [comment]
@ptierney 
Great! Thanks a lot for the effort!

I still reserve my personal opinion about:
"2). Can we decide the units based on the environment? Users may not know what the exact units for the geometries. Let's say if the geometries are in feet, what will it mean if users choose meters? "

Maybe we need a follow-up task to investigate this.

Other than this, LGTM!

**Comment by ptierney** [comment]
Yeah, there's been an ongoing set of discussions on units, I'll try to dig up some notes on these.

**Comment by ptierney** [comment]
This removes the wrong shared info file: https://github.com/DynamoDS/Dynamo/pull/4219

**Pull request merged**